collections/_posts/2022-11-28-CurriculumLearningByDIH.md

```diff
@@ -0,0 +1,110 @@
+  \right.
+$$
+
+$$ \gamma \in [0,1] $$ is a discount factor and $$S_t$$ is the set of samples used for training at time $$t$$.
+
+ Three instantaneous hardness are used :
+* the loss : $$l(y_i,F(x_i; w_t))$$, where $$l(·, ·)$$ is a standard loss function and $$F(·; w)$$ is the model where $$w$$ are the model parameters
+* the loss change between two consecutive time steps : $$ \|l(y_i,F(x_i; w_t))-l(y_i,F(x_i; w_{t-1}))\| $$
+* the prediction flip (the 0-1 indicator of whether the prediction correctness changes) between two consecutive time steps : $$ \| \mathbb{1} [ŷ^t_i = y_i]-\mathbb{1}[ŷ^{t-1}_i = y_i]\| $$, where $$ŷ^t_i$$ is the prediction of sample $$i$$ in step $$t$$, e.g., $$ argmax_j F(x_i; w_t)[j]$$ for classification.
+
+![](/collections/images/CL_DIH/DIH_vs_SPL.jpg)
+
+Compared to DIH instantaneous hardness is very less stable so as the training and it requires extra inference steps of a model over all the samples.
+
+
+## Properties
+
+DIH can vary a lot between different samples allowing their selection. When it is smaller, the samples are more memorable i.e. easy to learn while larger DIH means that it is harder to retain.
+
+![](/collections/images/CL_DIH/sample_separation.jpg)
+The curve shape is explained by a cyclic learning rate. We clearly see the gap between the training dynamics of red samples that are harder and blue samples that have smaller DIH. Moreover, the variance of sample with larger DIH is bigger showing that a local minima is not found for them and it is necessary to revisit them often, same conclusion regarding the prediction flip that is higher.
+
+![](/collections/images/CL_DIH/early_information.jpg)
+
+DIH in early epochs (40) suffices to identify the easy vs. the hard samples. It doesn't require a whole training to classify the samples. 
+DIH metrics decrease during training for both easy and hard samples indicating that as learning continues, samples become less informative, so that fewer samples can be selected for training.
+
+# Experiments
+
+## Curriculum learning
+
+As the DIH represents the hardness of samples it seems natural to base some curriculum learning on it (DIHCL). They keep training the model on samples whith large DIH that have historically been hard since the model does not perform well on them and revisit easy samples less frequently because the model is more likely to stay at those samples’ minima. 
+At each training step a subset of samples is selected according to their DIH values, given the probability $$p_{t,i} \propto h(r_{t-1}(i))$$ where $$h(·)$$ is a monotone non-decreasing function. 
+* DIHCL-Rand: data is sample proportionnaly to DIH $$h(r_t(i)) = r_t(i)$$
+* DIHCL-Exp: trade-off exploration/exploitation based on softmax value $$h(r_t(i)) = exp [\sqrt{2*log(n/n)}*r_t(i)], a_t(i) \leftarrow a_t(i)/p_{t,i} \forall i \in S_t$$
+* DIHCL-Beta: Beta prior distribution to balance exploration and exploitation $$h(r_t(i)) \leftarrow Beta(r_t(i),c-r_t(i))$$ with $$c > r_t(i)$$
```

Comment:
I would suggest making the function names bold to make them stand out a bit more.

Formatting + typo:
```suggestion
* **DIHCL-Rand**: data is sampled proportionally to DIH $$h(r_t(i)) = r_t(i)$$
* **DIHCL-Exp**: trade-off between exploration/exploitation based on softmax value $$h(r_t(i)) = exp [\sqrt{2*log(n/n)}*r_t(i)], a_t(i) \leftarrow a_t(i)/p_{t,i} \forall i \in S_t$$
* **DIHCL-Beta**: Beta prior distribution to balance exploration and exploitation $$h(r_t(i)) \leftarrow Beta(r_t(i),c-r_t(i))$$ with $$c > r_t(i)$$
```